packages/patterns/src/test/disposables.unit.ts

```diff
@@ -31,31 +31,51 @@ describe('disposables', () => {
         });
         it('times out when the disposal takes too long', async () => {
             const disposables = createDisposables();
-            disposables.add(
-                async () => {
+            disposables.add({
+                name: 'slow',
+                timeout: 1,
+                dispose: async () => {
                     await sleep(100);
                 },
-                { name: 'slow', timeout: 1 }
-            );
+            });
             await expect(disposables.dispose()).to.eventually.be.rejectedWith('Disposal timed out: "slow"');
         });
+        it('fail with the name of specific dispose', async () => {
+            const disposables = createDisposables();
+            disposables.add({
+                name: 'disposing with error',
+                dispose: () => {
+                    throw new Error('failed!');
+                },
+            });
+
+            try {
+                await disposables.dispose();
+                expect.fail('should have thrown');
+            } catch (e) {
+                const message = (e as Error).message;
+                expect(message, 'error message match pattern').to.matches(
+                    /Disposal failed: "disposing with error"\nError: failed!/
+                );
+            }
```

Comment:
You can use expect to throw

```suggestion
expect(
    async () => await disposables.dispose(), 
    'error message match pattern'
).to.throw(/Disposal failed: "disposing with error"\nError: failed!/);
```

Reply:
what about this code:
```js
expect(
    async () => await disposables.dispose(), 
    'error message match pattern'
).to.be.rejectedWith(/Disposal failed: "disposing with error"\nError: failed!/);
```